### PR TITLE
add trompeloeil

### PR DESCRIPTION
**Pull request by paul-reilly**
Adds the trompeloeil mocking library.

### CODE REVIEW

File: packages/t/trompeloeil/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    set_license("BSL-1.0")
+
+    add_urls("https://github.com/rollbear/trompeloeil/archive/refs/tags/$(version).tar.gz")
+    add_versions("v43", "86a0afa2e97347202a0a883ab43da78c1d4bfff0d6cb93205cfc433d0d9eb9eb")
+
+    on_install(function (package)
+        os.cp("include/*", package:installdir("include"))
+    end)
+
+    on_test(function (package)
```
Comment (waruqi) [inline packages/t/trompeloeil/xmake.lua:14]: too long test, can you simplify it?

File: packages/t/trompeloeil/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package("trompeloeil")
+
```
Comment (waruqi) [inline packages/t/trompeloeil/xmake.lua:2]: set_kind("library", {headeronly = true})

File: packages/t/trompeloeil/xmake.lua (hunk 1 above)
Comment (paul-reilly) [reply packages/t/trompeloeil/xmake.lua:14]: Yeah, I thought that might have been. No problem.

File: packages/t/trompeloeil/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    set_kind("library", {headeronly = true})
+    set_homepage("https://github.com/rollbear/trompeloeil")
+    set_description("A thread-safe header-only mocking framework for C++11/14 using the Boost Software License 1.0")
+    set_license("BSL-1.0")
+
+    add_urls("https://github.com/rollbear/trompeloeil/archive/refs/tags/$(version).tar.gz")
+    add_versions("v43", "86a0afa2e97347202a0a883ab43da78c1d4bfff0d6cb93205cfc433d0d9eb9eb")
+
+    on_install(function (package)
+        os.cp("include/*", package:installdir("include"))
```
Comment (waruqi) [inline packages/t/trompeloeil/xmake.lua:12]:
```
os.cp("include", package:installdir())
```

Comment (paul-reilly) [reply packages/t/trompeloeil/xmake.lua:12]: Okdk, will do.

**Pull request merged**